### PR TITLE
Add support for a log bucket

### PR DESCRIPTION
**Pull request by johncblandii**
## what

* Add support for access logs bucket

## why

* Consumers will not be required to create their own buckets anymore

## references

N/A

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -132,6 +133,26 @@ data "aws_iam_policy_document" "prevent_unencrypted_uploads" {
...
 
+module "log_storage" {
+  source  = "cloudposse/s3-log-storage/aws"
+  version = "0.37.1"
+
+  enabled                  = local.logging_bucket_enabled
+  acl                      = "log-delivery-write"
+  attributes               = ["logs"]
+  access_log_bucket_prefix = try(var.logging["prefix"], "logs/")
+  standard_transition_days = 30
```
Comment (nitrocode) [inline main.tf:144]: please expose these as input variables as they meet need to be configured based on design deciding

Comment (johncblandii) [reply main.tf]: Exposed those 3, but further customization would be done outside of this module so we don't copy all vars into it.

See bb20c72.

**Review by max-lobur — DISMISSED** [review]

File: main.tf [hunk 2]
```diff
@@ -132,6 +136,26 @@ data "aws_iam_policy_document" "prevent_unencrypted_uploads" {
...
+  name                     = local.logging_bucket_name_default
+  standard_transition_days = var.logging_bucket_standard_transition_days
+
+  context = module.this.context
+}
+
+locals {
+  logging_bucket_name = local.logging_bucket_enabled ? module.log_storage.bucket_id : local.logging_bucket_name_default
+  logging_prefix      = local.logging_bucket_enabled ? module.log_storage.prefix : local.logging_prefix_default
+}
```
Comment (nitrocode) [inline main.tf:157]: Can we keep these locals together at the top ?

File: main.tf [hunk 3]
```diff
@@ -176,8 +197,8 @@ resource "aws_s3_bucket" "default" {
   dynamic "logging" {
     for_each = var.logging == null ? [] : [1]
     content {
-      target_bucket = var.logging["bucket_name"]
-      target_prefix = var.logging["prefix"]
+      target_bucket = local.logging_bucket_name
```
[the remark this answers is not in the document]
Comment (nitrocode) [reply main.tf:203]: Not sure why this is being spammed every few hours... we should add kms encryption here as an option or default but maybe in a separate PR

Created this ticket for future ref: https://github.com/cloudposse/terraform-aws-tfstate-backend/issues/105

File: main.tf (hunk 2 above)
Comment (johncblandii) [reply main.tf:157]: Sure, I could. I put it there for sequencing purposes (exists after it is created), but I get it. Gimme a sec.

Comment (johncblandii) [reply main.tf]: Done in e8aaec4.

**Review by nitrocode — APPROVED** [review]

**Pull request merged**